src/hotspot/share/memory/allocation.hpp

```diff
@@ -151,14 +151,20 @@ class AllocatedObj {
 /*
  * Memory types
  */
-enum MemoryType {
+enum class MEMFLAGS {
```

Comment:
Remark (possible for a later change): As you wrote, I think using uint8 would be okay and safer here.